### PR TITLE
Only Unit Test Changed Modules

### PR DESCRIPTION
**Pull request by sethAmazon**
**Description:** 
Only unit test modules that have changed

**Link to tracking Issue:** 
#5928

**Testing:** 
Inside the test is testing code

### CODE REVIEW

File: unit_test_changed_modules_only_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+func TestRequiredDepsOnly(t *testing.T) {
+	changeFilesArray := getChangedFilesArray("git diff --name-only HEAD^..HEAD", t)
+	fileChangePathSet := getChangedPathSet(changeFilesArray, t)
+	moduleChangePathSet := getChangedModulesSet(fileChangePathSet, t)
+	pathSetOfModules := getAllModules(t)
+	modulesToTest := getAllModulesToTest(pathSetOfModules, moduleChangePathSet, t)
+	//test modules
+	for s := range modulesToTest {
+		out, err := exec.Command("bash", "-c", "cd " + s + "; go test ./... -v").Output()
```
Comment (Aneurysm9) [inline unit_test_changed_modules_only_test.go:35]:
```suggestion
		cmd := exec.Command("go", "test", "./...", "-v")
		cmd.Dir = filepath.Abs(s)
		out, err := cmd.Output()
```

File: unit_test_changed_modules_only_test.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		}
+		moduleChangePathSet[path] = true
+	}
+	for s := range moduleChangePathSet {
+		t.Log("Module Path Changed " + s)
+	}
+	return moduleChangePathSet
+}
+
+func getAllModules(t *testing.T) map[string]bool {
```
Comment (sethAmazon) [inline unit_test_changed_modules_only_test.go:129]: Replace with make file get all modules command

File: unit_test_changed_modules_only_test.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+func getAllModulesToTest(pathSetOfModules map[string]bool, moduleChangePathSet map[string]bool, t *testing.T) map[string]bool {
+	//get all modules to test
+	modulesToTest := make(map[string]bool)
+	for modPath := range pathSetOfModules {
+		for moduleChangePath := range moduleChangePathSet {
+			if modulesToTest[modPath] {
+				break
+			}
+			out, _ := exec.Command("bash", "-c",
```
Comment (sethAmazon) [inline unit_test_changed_modules_only_test.go:156]: Build a cache of go mod graph

File: unit_test_changed_modules_only_test.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+)
+
+func TestRequiredDepsOnly(t *testing.T) {
+	changeFilesArray := getChangedFilesArray("git diff --name-only HEAD^..HEAD", t)
+	fileChangePathSet := getChangedPathSet(changeFilesArray, t)
+	moduleChangePathSet := getChangedModulesSet(fileChangePathSet, t)
+	pathSetOfModules := getAllModules(t)
+	modulesToTest := getAllModulesToTest(pathSetOfModules, moduleChangePathSet, t)
+	//test modules
+	for s := range modulesToTest {
```
Comment (sethAmazon) [inline unit_test_changed_modules_only_test.go:34]: replace with using make test.

File: unit_test_changed_modules_only_test.go [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestRequiredDepsOnly(t *testing.T) {
```
Comment (sethAmazon) [inline unit_test_changed_modules_only_test.go:27]: write a list of modules to test with standard out then have the make file read those modules and test

**Pull request closed without merging**